pac4j-core/src/main/java/org/pac4j/core/client/DirectClientV2.java

```diff
@@ -38,7 +39,14 @@ protected C retrieveCredentials(final WebContext context) throws HttpAction {
             this.authenticator.validate(credentials, context);
             return credentials;
         } catch (CredentialsException e) {
-            logger.error("Failed to retrieve or validate credentials", e);
+            if (logger.isDebugEnabled()) {
```

Comment:
Well... do we agree @mmoayyed that it is what you had in mind:
```java
            logger.debug("Failed to retrieve or validate credentials", e);
            logger.info("Failed to retrieve or validate credentials: {}", e.getMessage());
```
Or was it:
```java
            if (logger.isDebugEnabled()) {
                logger.debug("Failed to retrieve or validate credentials", e);
            } else {
                logger.info("Failed to retrieve or validate credentials: {}", e.getMessage());
            }
```

In the first case it feels really strange to get two logs saying almost the same thing and in the second case, it's enough to have DEBUG logs redirected to another place than INFO logs (in the logs configuration) for the information to be easily missed...

Please tell me what you prefer :)